examples/ewald_total.cpp

```diff
@@ -482,27 +484,28 @@ std::vector<Real> evaluate_short_range(const TestCaseSystem<Real> &System, Short
 
     // shortcuts
     const int n_boxes = Short.n_boxes;
-    const int n_dim = System.n_dim;
     const int n_trg = System.n_trg;
-    const int n_src = System.n_src;
-    const Real L = System.L;
-
     // std::cout << "num boxes = " << n_boxes << " num dims = " << n_dim << " num neighbors = " <<
     // Short.box_neighbors.size() << "num box corners = " << Short.box_corners.size() << std::endl;
-    const Real r_cut_sq = r_cut * r_cut;
     std::vector<Real> pot(n_trg, 0.0);
     std::vector<Real> pot_sorted(n_trg, 0.0); // sorted potential vector
-    std::vector<Real> offset(n_dim, 0.0);
 
-    // iterate through all boxes
+// iterate through all boxes
+#pragma omp parallel for default(none) shared(n_boxes, pot_sorted, System, r_cut, Short, alpha, vectorized, n_trg)
     for (size_t box = 0; box < n_boxes; ++box) {
         // go through the neighbors
+        const Real r_cut_sq = r_cut * r_cut;
+        const int n_dim = System.n_dim;
+        const int n_src = System.n_src;
+        const Real L = System.L;
+
         const Real *x = &(Short.r_src_sorted[Short.box_begin[box]]);
         const Real *y = &(Short.r_src_sorted[n_src + Short.box_begin[box]]);
         const Real *z = &(Short.r_src_sorted[n_src * 2 + Short.box_begin[box]]);
 
         Real *pot_part = &(pot_sorted[0]) + Short.box_begin[box];
         const int *box_corner = &Short.box_corners[box * n_dim];
+        std::vector<Real> offset(n_dim, 0.0);
```

Comment:
eventually we just template on n_dim, so `std::array<Real, n_dim>` will be good as it's guaranteed to be a stack allocation (WAY cheaper than std::vector which will default to heap without a custom allocator). I'd just `constexpr n_dim = 3` now if it's not already a template argument.